api/v1/notification_types.go

```diff
@@ -76,6 +76,10 @@ type NotificationSpec struct {
 	// WaitForEvalPeriod is an additional delay after WaitFor before evaluating
 	// Kubernetes config health. Format: "5m", "1h"
 	WaitForEvalPeriod *string `json:"waitForEvalPeriod,omitempty" yaml:"waitForEvalPeriod,omitempty"`
+
+	// WaitForGroup allows notifications in waiting status to be grouped together
+	// based on certain set of keys.
+	WaitForGroup []string `json:"waitForGroup,omitempty"`
```

Comment:
This should probably be groupBy with an error thrown if groupBy is specified without a waitFor